### PR TITLE
(feat) support preserveValueImports introduced in TS 4.5

### PR DESCRIPTION
**Pull request by dummdidumm**
This new flag keeps imports as is, essentially functioning the same as the import transformer.

### Before submitting the PR, please make sure you do the following

- [ ] It's really useful if your PR relates to an outstanding issue, so please reference it in your PR, or create an explanatory one for discussion. In many cases features are absent for a reason.
- [x] This message body should clearly illustrate what problems it solves. If there are related issues, remember to reference them.
- [x] Ideally, include a test that fails without this PR but passes with it. PRs will only be merged once they pass CI. (Remember to `npm run lint`!)

### Tests

- [x] Run the tests with `npm test` or `yarn test`

### CODE REVIEW

**Review by bluwy — APPROVED** [review]

File: test/transformers/typescript.test.ts [hunk 1]
```diff
@@ -140,6 +140,23 @@ describe('transformer - typescript', () => {
...
+
+      const opts = sveltePreprocess({
+        typescript: {
+          tsconfigFile: false,
+          compilerOptions: { preserveValueImports: true },
+        },
+      });
+
+      const { code } = await preprocess(tpl, opts);
+      expect(code).toContain("import { Bar } from './somewhere'");
```
Comment (kaisermann) [inline test/transformers/typescript.test.ts:154]:
```suggestion

      expect(code).toContain("import { Bar } from './somewhere'");
```

**Review by kaisermann — APPROVED** [review]

**Comment by kaisermann** [comment]
@dummdidumm  LGTM, but I'm a bit out of context and busy with moving to another country, sorry for taking some days to check this! This is possible due to TS 4.5, right?

**Comment by dummdidumm** [comment]
Correct. TS 4.5 introduced the new setting `preserveValueImports` along with DX to neatly separate types and values. You can read more on that here: https://devblogs.microsoft.com/typescript/announcing-typescript-4-5/#preserve-value-imports

File: package.json [hunk 2]
```diff
@@ -87,7 +87,7 @@
     "sugarss": "^2.0.0",
     "svelte": "^3.42.0",
     "ts-jest": "^25.1.0",
-    "typescript": "^4.4.2"
+    "typescript": "^4.5.2"
```
Comment (kaisermann) [inline package.json:90]: Quick question: (when) have we removed the support for v3? 🤔

Comment (dummdidumm) [reply package.json:90]: never, these are the dev dependencies, not the peer dependencies.

**Pull request merged**